### ISSUE TITLE
Outdated croniter dependency

### ISSUE THREAD

**Issue by cuu508** [apache/airflow#23882]
### Apache Airflow version

2.3.0 (latest released)

### What happened

setup.cfg contains 

    croniter>=0.3.17

croniter versions before 1.0.5 have a security issue: if you pass expressions
like "0-1000000000 * * * *" to it, croniter can throw an exception, or
take a long time to return, or crash the python process.

### What you think should happen instead

It would be a good idea to tighten up the croniter dependency and require at least version 1.0.5. 

### Operating System

N/A

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by eladkal** [comment]
Airflow is recommended to be installed with [constraints](https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#installation-tools)

If you follow the guidelines and check the constraints file you will see:
https://github.com/apache/airflow/blob/d18aee02142e0ab285101c45c60de1c81ef7e7ad/constraints-3.7.txt#L206

I have no objection for setting higher lower bound but for the end user who follow documentation this is not an issue.

**Comment by uranusjr** [comment]
IMO Airflow should not be eagerly determining what version of a package should be used here. Any croniter versions newer than 0.3.17 is compatible, and that’s what Airflow’s dependency metadata should care about.

**Comment by potiuk** [comment]
We can add lower-bound limitations - and we did it for some really serious security issues when we wanted to make sure that our users did not use certain dependencies (or where we new there is a compatibility issue. But in this case, where DAG writer can anyhow do whatever (including spawning millions of processes or allocating all memory or deleting the whole database, the fact that they can also write a potentialy crashing cron expression, is basically no issue.
Closing.
